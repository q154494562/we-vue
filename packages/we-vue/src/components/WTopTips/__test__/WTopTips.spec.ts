import { createLocalVue, mount, shallowMount } from '@vue/test-utils'
import TopTipsApi from '../'
import TopTips from '../WTopTips'

describe('test top-tips api', () => {
  beforeEach(() => {
    jest.useFakeTimers()
  })

  afterEach(() => {
    TopTipsApi.close()
    jest.clearAllTimers()
  })

  test('open a top-tips and then close it', () => {
    const instance = TopTipsApi({
      message: 'test',
      duration: 3000,
    })

    jest.advanceTimersByTime(1000)

    TopTipsApi.close()

    expect(instance.visible).toBe(false)
  })

  test('open a top-tips with string parameter', () => {
    const localVue = createLocalVue()
    const instance = TopTipsApi('test')

    expect(instance.visible).toBe(true)

    localVue.nextTick(() => {
      expect(document.querySelectorAll('.weui-toptips')).toHaveLength(1)
      expect(document.querySelector('.weui-toptips')!.textContent).toBe('test')
    })
  })

  test('create a toast with duration', () => {
    const instance = TopTipsApi({
      message: 'foo',
      duration: 2000,
    })

    jest.advanceTimersByTime(2001)

    expect(instance.visible).toBe(false)
  })

  test('top-tips should be singletom', () => {
    const localVue = createLocalVue()
    TopTipsApi({
      message: 'foo',
    })

    localVue.nextTick(() => {
      // try to open another top-tips
      TopTipsApi({
        message: 'foo',
      })

      expect(document.querySelectorAll('.weui-toptips')).toHaveLength(1)
    })
  })

  test('setDefaultOptions and resetDefaultOptions', () => {
    const theDefault = {
      visible: true,
      message: '',
      duration: 3000,
    }

    const opts = {
      duration: 3000,
    }

    TopTipsApi.setDefaultOptions(opts)

    expect(TopTipsApi.defaultOptions).toEqual({ ...theDefault, ...opts })

    TopTipsApi.resetDefaultOptions()

    expect(TopTipsApi.defaultOptions).toEqual(theDefault)
  })
})

describe('top-tips component', () => {
  test('create', () => {
    const wrapper = mount(TopTips, {
      attachToDocument: true,
      propsData: {
        visible: true,
      },
    })

    expect(wrapper.name()).toBe('w-top-tips')
    expect(wrapper.classes()).toContain('weui-toptips')
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should render message correctlly', () => {
    const wrapper = shallowMount(TopTips, {
      propsData: {
        message: 'test',
        visible: true,
      },
    })

    expect(wrapper.text()).toContain('test')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
