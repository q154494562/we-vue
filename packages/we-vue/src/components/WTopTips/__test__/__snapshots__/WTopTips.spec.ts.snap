// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`top-tips component create 1`] = `<div class="weui-toptips weui-toptips_warn"></div>`;

exports[`top-tips component should render message correctlly 1`] = `
<transition-stub enteractiveclass="weui-animate-fade-in" leaveactiveclass="weui-animate-fade-out">
  <div class="weui-toptips weui-toptips_warn">test</div>
</transition-stub>
`;
